synonyms/linter.js

```diff
@@ -74,6 +75,12 @@ function multiWordCheck(line, logprefix, tokens) {
   });
 }
 
+function tokenReplacementCheck(line, logprefix) {
```

Comment:
I've also added this additional linter warning.

Technically this syntax is actually still supported (the linter doesn't rewrite the configs) but I would like to heavily discourage its use.

The reason for this is an entry such as `road => rd` would only enter `[ "rd" ]` in the index, so when a user queries for `[ "road" ]` (or any ngram of) it will not match.

Advanced users may opt to write it as `road => road, rd` to overcome this issue but this is equivalent to `road, rd`.

I think we will probably completely drop support for `=>` at some point in the future because it's very easy to make make mistakes with.

Reply:
I was thinking it might actually be useful to use `=>` sometimes. For example, if we have a single letter abbreviation that we want to ensure is not being applied too often.

`c, carrer` would replace every instance of `c` with `carrer`. However, `carrer => career, c` would not.

As you mentioned we would probably recommend that all tokens on the left hand side of the replacement also appear on the right, to avoid issues with autocomplete jitter.